modules/enet/networked_multiplayer_enet.cpp

```diff
@@ -377,16 +374,37 @@ void NetworkedMultiplayerENet::close_connection() {
 
 	if (peers_disconnected) {
 		enet_host_flush(host);
-		OS::get_singleton()->delay_usec(100); //wait 100ms for disconnection packets to send
+		OS::get_singleton()->delay_usec(100); // Wait 100ms for disconnection packets to send
 	}
 
 	enet_host_destroy(host);
 	active = false;
 	incoming_packets.clear();
-	unique_id = 1; //server is 1
+	unique_id = 1; // Server is 1
 	connection_status = CONNECTION_DISCONNECTED;
 }
 
+void NetworkedMultiplayerENet::disconnect_peer(int p_peer, bool now) {
+
+	ERR_FAIL_COND(!active);
+	ERR_FAIL_COND(!is_server());
+	ERR_FAIL_COND(p_peer <= 1);
+
+	Map<int, ENetPeer *>::Element *E = NULL;
+
+	E = peer_map.find(ABS(p_peer));
+	if (!E) {
+		ERR_EXPLAIN("Invalid peer: " + itos(p_peer));
+		ERR_FAIL();
+	}
+
+	if (now) {
+		enet_peer_disconnect_now(E->get(), 0);
+	} else {
+		enet_peer_disconnect_later(E->get(), 0);
```

Comment:
You can use `peer_map[p_peer]` to retrieve the peer directly without the find (since you will be checking it with `has` in the guard)